logging_loki/handlers.py

```diff
@@ -86,3 +83,38 @@ def emit(self, record: logging.LogRecord):
             self.emitter(record, self.format(record))
         except Exception:
             self.handleError(record)
+
+    def emit_batch(self, records: list[logging.LogRecord]):
+        """Send a batch of log records to Loki."""
+        # noinspection PyBroadException
+        try:
+            self.emitter.emit_batch([(record, self.format(record)) for record in records])
+        except Exception:
+            for record in records:
+                self.handleError(record)
+
+class LokiBatchHandler(MemoryHandler):
+    interval: float # The interval at which batched logs are sent in seconds
+    _last_flush_time: float
+    target: LokiHandler
+
+    def __init__(self, interval: float, capacity: int = LOKI_MAX_BATCH_BUFFER_SIZE, **kwargs):
+        super().__init__(capacity, **kwargs)
+        self.interval = interval
+        self._last_flush_time = time.time()
+
+    def flush(self) -> None:
+        self.acquire()
+        try:
+            if self.target and self.buffer:
+                self.target.emit_batch(self.buffer)
+                self.buffer.clear()
+        finally:
+            self.release()
+        self._last_flush_time = time.time()
```

Comment:
If we have an error when emitting the logs batch or clearing the buffer, we don't want to set the last flush time?

Reply:
yeah sure, would be better to always update it indeed